### PR TITLE
Run `clippy --fix -- -D clippy::pedantic`

### PR DESCRIPTION
**Pull request by Kijewski**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
CI is unhappy. Also, do you think it's worth having it in CI?

**Comment by Kijewski** [comment]
> CI is unhappy.

Yeah, I added some `#[must_use]`s manually, and clippy is unhappy if the return type is already `#[must_use]`. Fixed.

> Also, do you think it's worth having it in CI?

There is still a ton of pedantic warnings. Only the automatically fixable ones are changed by this PR. IMHO many warnings in `clippy::pedantic` are, well, overly pedantic. :) But I guess we can gradually enable more clippy lints. E.g. from a users' point-of-view [`missing_errors_doc`](https://rust-lang.github.io/rust-clippy/rust-1.81.0/index.html#/missing_errors_doc) is very useful, but has to be done manually.

File: rinja_derive/src/input.rs [hunk 1]
```diff
@@ -93,7 +93,8 @@ impl TemplateInput<'_> {
 
         let escaping = escaping
             .as_deref()
-            .unwrap_or_else(|| path.extension().map(|s| s.to_str().unwrap()).unwrap_or(""));
+            .or_else(|| path.extension().and_then(|s| s.to_str()))
+            .unwrap_or_default();
```
Comment (Kijewski) [inline rinja_derive/src/input.rs:97]: This change was done manually.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**